utils/encryption/encryption.go

```diff
@@ -0,0 +1,126 @@
+package encryption
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+
+	"golang.org/x/crypto/nacl/box"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+)
+
+const KeySize = 32
+
+var (
+	errKeySize = fmt.Errorf("%w: recipient key has invalid size", commonerrors.ErrInvalid)
+)
+
+type keyPair struct {
+	public  string
+	private string
```

Comment:
I know you shouldn't use pointers to primitives but would it be worth having these (at least the private key) as pointers so that doing `Log`/`Printf`/`Println` doesn't expose the key? Even though they aren't exposed and the `keyPair` isn't exposed, you can still print the resulting `IKeyPair` and see the contents. Wity pointers you won't get the values themselves.

I know it doesn't completely solve the problem as you can use the "unsafe" module and get the value at the pointer, but it might help prevent leaking information in logs. 

Whether it's worth the effort to do this, is up to you.